### PR TITLE
build!: update dependencies

### LINKED ISSUE #205: Unpin exact versions for dependencies
This issue is blocked until all normalizers are updated on main branch and a version is released. In the main branch, we are unpinning exact versions for dependencies. 

In main branch:
- [ ] Update disease normalizer version to ~= 0.2.20 
- [ ] Update gene normalizer version to ~= 0.1.35
- [ ] Update variation normalizer version to ~= 0.5.4
- [ ] Update therapy normalizer version to ~= 0.3.10
- [ ] Update civicpy version to ~=3.0.0
- [ ] Update ga4gh.vrs to ~=0.8.1
- [ ] Update ga4gh.vrsatile.pydantic to ~=0.0.12

### PR DESCRIPTION
**Pull request by korikuzma**
close #205

Notes:
- We never made a pypi release for 1.1.1. I can bump the version (minor or patch) if suggested
- Updated dependencies. Tried to not make any additional changes since I knew this PR was going to be large. 
- `Harvester` base class
  - Remove instance var `assertions`
  - Create timestamped harvester json if no filename given (now includes empty string)
- `CIViCHarvester` 
  - Major cleanup. We are not picky about what fields we get now. We will only exclude civic fields that start with `_` or `partial`  
  - We now harvest civic molecular profiles
  - We no longer get `civic_actionability_score` and `variant_groups`. I couldn't find `civic_actionability_score`, but maybe I am missing it. We didn't really use `variant_groups` in our searches, so I figure we can add back if needed in a new issue. 
- `OncoKBHarvester`
  - Use enum values for api endpoints 
- `metakb.main`
  - Use Path for `/api/v2/search/{id}` rather than Query
- `metakb.normalizers`
  - Rename instance var `variation_normalizer` --> `variation_query_handler`  
  - Allow for query handler instances to be passed as parameters
- `CIViCTransform` : Updates from civicpy 3.0
  - Include molecular profile ID to variant ID mapping
  - Assertion's now have `assertion_type`
  - `drugs` --> `therapies`
- `OncoKBTransform`
  - `uuid` for drug returned `None`, so switched to using `ncitCode` for the therapy descriptor mappings
- `setup.cfg`
  - license-file --> license_file in setup.cfg. See [here](https://setuptools.pypa.io/en/latest/history.html#id510)  
- Tests
  - In test_database.py::test_variation_rules, I removed a check since we now support copy number variation from {GENE} Amplification queries
  - Updated tests to reflect source changes 
- `metakb.cli`
  - Added `--update_from_remote` for civicpy cache 
  - I ran the harvesters and transformers using the prod ddbs. See the `vicc-metakb` s3 bucket from outputs on `20230507`. Displaying the logs / CLI output here to later reference.
    ```
    Harvesting civic...
    (civicpy cache is also being updated)
    civic harvest finished in 222.36153 s
    Harvesting moa...
    moa harvest finished in 4.87718 s
    Harvesting oncokb...
    oncokb harvest finished in 2073.78965 s
    Successfully harvested all sources in 2301.02900 s
    
    Transforming harvested data to CDM...
    Transforming civic...
    civic transform finished in 893.77244 s.
    Transforming moa...
    moa transform finished in 1558.83657 s.
    Transforming oncokb...
    oncokb transform finished in 3776.59320 s.
    Successfully transformed all sources to CDM in 6230.36656 s

    ...

    [2023-05-07 18:00:08,141] - metakb.database - INFO : Loading data from metakb/data/civic/transform/civic_cdm_20230507.json
    [2023-05-07 18:01:14,634] - metakb.database - INFO : Successfully loaded 835 statements.
    [2023-05-07 18:01:14,638] - metakb.database - INFO : Loading data from metakb/data/moa/transform/moa_cdm_20230507.json
    [2023-05-07 18:01:27,687] - metakb.database - INFO : Successfully loaded 176 statements.
    [2023-05-07 18:01:27,688] - metakb.database - INFO : Loading data from metakb/data/oncokb/transform/oncokb_cdm_20230507.json
    [2023-05-07 18:08:43,168] - metakb.database - INFO : Successfully loaded 6056 statements. 
  ```

### CODE REVIEW

File: metakb/version.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 """MetaKB version"""
 # REQ: EACH TIME VERSION IS UPDATED, MUST ALSO UPDATE LAST_UPDATED
 __version__ = "1.1.1"
```
Comment (jsstevenson) [inline metakb/version.py:3]: maybe too late to start doing this now but we probably should've been using a 'dev' or 'alpha' marker in the version numbering

**Review by jsstevenson — DISMISSED** [review]
👍

File: metakb/version.py (hunk 1 above)
Comment (korikuzma) [reply metakb/version.py:3]: We could bump to `1.2.0-dev0`

**Comment by korikuzma** [comment]
Forgot to update gene-normalizer 🤦‍♀️  @jsstevenson is it ok if i make a new release? See changes [here](https://github.com/cancervariants/gene-normalization/compare/0.1.35...main)

File: metakb/version.py (hunk 1 above)
Comment (korikuzma) [reply metakb/version.py:3]: @jsstevenson let me know if you want 1.1.1, 1.2.0-dev0, or something else. I think the minor bump makes sense

Comment (jsstevenson) [reply metakb/version.py:3]: @korikuzma whatever works! I was just thinking that it's not really released yet (or is it???) so numbering might need to reflect that

Comment (korikuzma) [reply metakb/version.py:3]: Makes sense, I'll switch to 1.2.0-dev0

**Pull request merged**